testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/realm/RealmRolesTest.java

```diff
@@ -257,5 +257,43 @@ public void roleMembershipAfterUserRemoval() {
         assertEquals(0, roleResource.getRoleUserMembers().size());
 
     }
-    
+
+    /**
+     *
+     */
+    @Test
+    public void testRoleMembershipWithPagination() {
+        RoleResource role = resource.get("role-with-users");
+
+        // Add a second user
+        UserRepresentation userRep2 = new UserRepresentation();
+        userRep2.setUsername("test-role-member2");
+        userRep2.setEmail("test-role-member2@test-role-member.com");
+        userRep2.setRequiredActions(Collections.<String>emptyList());
+        userRep2.setEnabled(true);
+        adminClient.realm(REALM_NAME).users().create(userRep2);
+
+        List<UserRepresentation> users = adminClient.realm(REALM_NAME).users().search("test-role-member", null, null, null, null, null);
+        assertEquals(2, users.size());
+        for (UserRepresentation userRepFromList : users) {
+            UserResource user = adminClient.realm(REALM_NAME).users().get(userRepFromList.getId());
+            UserRepresentation userRep = user.toRepresentation();
+
+            RoleResource roleResource = adminClient.realm(REALM_NAME).roles().get(role.toRepresentation().getName());
+            List<RoleRepresentation> rolesToAdd = new LinkedList<>();
+            rolesToAdd.add(roleResource.toRepresentation());
+            adminClient.realm(REALM_NAME).users().get(userRep.getId()).roles().realmLevel().add(rolesToAdd);
+        }
+
+        RoleResource roleResource = adminClient.realm(REALM_NAME).roles().get(role.toRepresentation().getName());
+        Set<UserRepresentation> roleUserMembers = roleResource.getRoleUserMembers(0, 1);
+        assertEquals(roleUserMembers.size(), 1);
+
+        roleUserMembers = roleResource.getRoleUserMembers(1, 1);
+        assertEquals(roleUserMembers.size(), 1);
```

Comment:
In addition, please check a detail of the users, not only size of the result list, since the two calls (lines 290, 293) could potentially return the same user due to some bug - and we want to catch it here.

Also prefer `assertThat(roleUserMembers, hasSize(1));` to `assertEquals`. (similarly on l. 277, 290)